virttest/utils_misc.py

```diff
@@ -2370,6 +2370,64 @@ def get_qemu_best_cpu_model(params):
     return params.get("default_cpu_model", None)
 
 
+def get_qemu_version(params):
+    """
+    Get the qemu-kvm(-rhev) version info.
+
+    :param params: Passed to get_qemu_binary, can set to {} if not needed.
+    :return: A dict contain qemu versoins info as {'major': int, 'minor': int,
+    'update': int, 'is_rhev': bool}
+    """
+    version = {'major': None, 'minor': None, 'update': None, 'is_rhev': False}
+    regex = r'\s*[Ee]mulator [Vv]ersion\s*(\d+)\.(\d+)\.(\d+)'
+
+    qemu_binary = get_qemu_binary(params)
+    version_raw = commands.getoutput("%s -version" % qemu_binary).splitlines()
+    for line in version_raw:
+        search_result = re.search(regex, line)
+        if search_result:
+            version['major'] = int(search_result.group(1))
+            version['minor'] = int(search_result.group(2))
+            version['update'] = int(search_result.group(3))
+        if "rhev" in str(line).lower():
+            version['is_rhev'] = True
+    for key, value in version.iteritems():
+        if value is None:
+            logging.error("Cannot fully get version info of the local qemu.")
+            break
+    return version
+
+
+def compare_qemu_version(major, minor, update, is_rhev=True, params={}):
+    """
+    Check if local install qemu versions is newer than provided version.
+
+    :param major: The major version to be compared.
+    :param minor: The minor version to be compared.
+    :param update: The update version to be compared.
+    :param is_rhev: If the qemu is a rhev version.
+    :param params: Other params.
+    :return: True means local installed version is equal or newer than the
+    version provided, otherwise False will be returned.
+    """
+    installed_version = get_qemu_version(params)
+    for key, value in installed_version.iteritems():
+        if value is None:
+            logging.error("Qemu version not clear, return False directly.")
+            return False
```

Comment:
This can be done in get_qemu_version(), return None in L2397 after logging will be more appropriate.

Reply:
fixed